Add tests for ProjectedRow, DeltaRecord and StorageUtil

yangjuns opened the pull request.

mbutrovich commented:
You'll have to rebase with @pervazea's changes today. It shouldn't be too bad from a conflict standpoint. Any test fixtures you modified or added will probably just need to inherit from the new test harness.

tli2 commented on test/storage/delta_record_test.cpp:
```diff
@@ -0,0 +1,126 @@
+
+namespace terrier {
+
+struct DeltaRecordTests : public ::terrier::TerrierTest {
```
nit: you are already under terrier namespace so `struct DeltaRecordTests : public TerrierTest` would suffice

tli2 commented on test/storage/projected_row_test.cpp:
```diff
@@ -0,0 +1,110 @@
+
+namespace terrier {
+
+struct ProjectedRowTests : public ::terrier::TerrierTest {
```
same nit

tli2 commented on test/storage/projected_row_test.cpp:
```diff
@@ -0,0 +1,110 @@
+      lower_bound = update->AccessForceNotNull(static_cast<uint16_t >(col - 1));
+      upper_bound = StorageTestUtil::IncrementByBytes(update->AccessForceNotNull(static_cast<uint16_t >(col - 1)),
+                                                      layout.attr_sizes_[col]);
+      // check if the value address is in bound
```
not in bound?

The use of CheckNotInBounds is probably correct judging by your comment outside the loop, but maybe it's most straightforward to instead advance the lower bound incrementally and use CheckInBounds against a constant upper bound.

yangjuns replied:
I saw that was what you did for TAS test. My concern was that if you fix the upper bound to be the very end, then the bound is too big and the address can still be incorrect even if it's in bound.  
On second thought, I think your way of checking bound is still correct because of some sort of "induction" reason. 

Correct me if I'm wrong.

yangjuns replied:
What I'm doing here is to,  for each column value,  identify the memory space used by the previous value, which is [lower_bound, upper_bound) and make sure that the current value's memory address don't start within this range.

yangjuns replied:
Never mind. I agree that advancing the lower bound and keeping upper bound constant serve the same purpose and are more straightforward.

tli2 commented on test/storage/storage_util_test.cpp:
```diff
@@ -0,0 +1,211 @@
+
+namespace terrier {
+
+struct StorageUtilTests : public ::terrier::TerrierTest {
```
same nit

yangjuns commented:
Quick Summary: This PR

- added 3 test files for testing ProjectedRow, DeltaRecord and StorageUtil.
- removed redundant namespace in `tuple_access_strategy_test.cpp`, `data_table_test.cpp`, `data_table_concurrent_test.cpp` as pointed out by @tli2
